### DISCUSSION TITLE
BarrierFunctionMaterial with coefficients

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#20667] [Q&A General] [answered]
Hi!

I am in the process of learning MOOSE, but I have a question.

Using MOOSE BarrierFunctionMaterial with g_order high, (g(η) = η^2(1−η ^2) ^2). Is it possible, in a straightforward way, to add coefficients in front of the η-term, where one of the terms needs to be space-depended?

All help and explanation are very much appreciated.

Best regards
Karl

**Comment by GiudGiud** [comment, marked as answer]
Hello

I looked at the parameters and it doesnt look like it.
I suggest you use a ParsedMaterial to add this prefactor:
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html

Guillaume

**Reply by MScKarl** [reply]
Thank you for your answer. If I’ve understood correctly, then I can define it as I’ve done it below, I just need to rewrite the math function into a MOOSE function? But how do I define spatial dependence of the order parameter that looks like:  g/2(∇η)^2 as a coefficient? 
```
[./f_eta]
  type = ParsedMaterial
  f_name = f_eta # the name of the function
  args = 'alpha beta gamma' # here is where a define the coefficients
  function =  Ft[η] = ∫[ g/2(∇η)^2 + (α/2)η^2 + (β/4)η^4+ (γ/6)η^6 ]dr

  outputs = exodus
[../]
```
I apologize if my questions are somewhat strange or trivial, but I’m very new to MOOSE, but excited to learn. 

Thanks again for all the help.

**Reply by GiudGiud** [reply]
Hello

Please see the discussion in this post for how to get gradients of material properties. 
https://github.com/idaholab/moose/discussions/20628
Basically you will have to propagate derivatives, starting from the gradients of the primary variables. 

Then once you have defined a gradient of a material property you can use it in a ParsedMaterial as part of the prefactor

Guillaume
